### PR TITLE
add adjustable height textview

### PR DESCRIPTION
**Pull request by unpluggedk**
Add a new field - Adjustable height multiline textview
Also added was the example to include this new field.

### CODE REVIEW

File: PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m [hunk 1]
```diff
@@ -127,6 +127,11 @@ - (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range r
     return YES;
 }
 
+- (void)textViewDidChange:(UITextView *)textView{
+    if(_isAdjustable)
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m]: Please always use curly parentheses around blocks.

File: PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m [hunk 2]
```diff
@@ -148,4 +153,45 @@ - (void)adjustSizeToContentWithMaxHeight:(CGFloat)maxHeight {
     self.view.frame = cellRect;
 }
 
+- (void)changeHeight:(BOOL)animated{
+
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m]: Please remove the empty line.

File: PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m [hunk 3]
```diff
@@ -148,4 +153,45 @@ - (void)adjustSizeToContentWithMaxHeight:(CGFloat)maxHeight {
...
+            
+            frame = self.view.frame;
+            frame.size.height = newHeight;
+            self.view.frame = frame;
+            
+            [[NSNotificationCenter defaultCenter] postNotificationName:PAYFormRowHeightChangeEndNotification object:nil];
+        }
+        
+    }
+
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m]: Please remove empty lines

File: PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m [hunk 4]
```diff
@@ -148,4 +153,45 @@ - (void)adjustSizeToContentWithMaxHeight:(CGFloat)maxHeight {
...
+            frame.size.height = newHeight;
+            self.textView.frame = frame;
+            
+            frame = self.view.frame;
+            frame.size.height = newHeight;
+            self.view.frame = frame;
+            
+            [[NSNotificationCenter defaultCenter] postNotificationName:PAYFormRowHeightChangeEndNotification object:nil];
+        }
+
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/ViewHolder/PAYFormMultiLineTextField.m]: -- empty line

File: PAYFormBuilder/controller/builder/PAYFormSectionBuilder.m [hunk 5]
```diff
@@ -167,6 +191,8 @@ - (PAYFormMultiLineTextField *)addTextViewWithPlaceholder:(NSString *)placeholde
     return formField;
 }
 
+
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/builder/PAYFormSectionBuilder.m]: --empty lines

File: PAYFormBuilder/controller/builder/PAYFormSectionBuilder.m [hunk 6]
```diff
@@ -136,27 +136,51 @@ - (PAYFormMultiLineTextField *)addTextViewWithPlaceholder:(NSString *)placeholde
...
     formField.name     = self.section.name;
+
+    if(isAdjustable){
+        formField.isAdjustable = YES;
+        formField.textView.scrollEnabled = NO;
+    }
+    else{
+        formField.isAdjustable = NO;
+        formField.textView.scrollEnabled = YES;
+    }
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/builder/PAYFormSectionBuilder.m]: This statement could be simplified in the following way:
```
formField.isAdjustable = isAdjustable;
formField.textView.scrollEnabled = !isAdjustable;
```

**Comment by Eldorado234** [comment]
Looks very good to me. Thank for the PR. I just added some notes regarding code style (partly my mistake). 

One think that could be optimised, if you like: The cell and the table are not animated synchronously. I found a solution for this problem in the following thread: http://stackoverflow.com/questions/460014/can-you-animate-a-height-change-on-a-uitableviewcell-when-selected The `targetHeight` could be stored in the ViewHolder and could be returned in https://github.com/Eldorado234/PAYFormBuilder/blob/master/PAYFormBuilder/controller/PAYFormTableViewController.m#L145

For me it is not a must have to merge this PR. Just leave me a note, if you like to add this. Thanks.

**Comment by unpluggedk** [comment]
Thanks, I've applied the code styling per your instructions.
At the moment, I think it's cool without optimizations. :+1:

**Pull request merged**